src/VideoManager/SubtitleWriter.cc

```diff
@@ -33,13 +35,35 @@ SubtitleWriter::SubtitleWriter(QObject* parent)
 
 void SubtitleWriter::startCapturingTelemetry(const QString& videoFile)
 {
-    // Get the facts displayed in the values widget and capture them, removing the "Vehicle." prefix.
-    QSettings settings;
-    settings.beginGroup("ValuesWidget");
-    _values = settings.value("large").toStringList().replaceInStrings(QStringLiteral("Vehicle."), QString());
-    _values += settings.value("small").toStringList().replaceInStrings(QStringLiteral("Vehicle."), QString());
+    QString factNameTemplate = QStringLiteral("rows/%1/columns/%2/factName");
+    QString groupNameTemplate = QStringLiteral("rows/%1/columns/%2/groupName");
+
+    // Delete facts of last run
+    _facts.clear();
+
+    // check user settings first, use default if user settings are empty
+    for (const QString& settingsGroup : settingsSources) {
+        QSettings settings;
+        settings.beginGroup(settingsGroup);
+        const int nRows = settings.value("rows/size").toInt();
+        // iterate through all rows and columns saving the facts into _facts
+        for(int row = 1; row < nRows + 1; row++) {
+            int nColumns = settings.value(QString("rows/%1/columns/size").arg(row)).toInt();
```

Comment:
Maybe you should check if the toInt calls are working and settings.value got a non invalid qvariant and print an warning message, failing in silence is terrible to debug.